maintainer/CI/fix_style.sh

```diff
@@ -18,36 +18,19 @@
 
 cd "$(git rev-parse --show-toplevel)"
 
-CLANG_FORMAT_VER=6.0
-AUTOPEP8_VER=1.3.4
-PYCODESTYLE_VER=2.3.1
-
 if ! git diff-index --quiet HEAD -- && [ "${1}" != "-f" ]; then
     echo "Warning, your working tree is not clean. Please commit your changes."
     echo "You can also call this script with the -f flag to proceed anyway, but"
     echo "you will then be unable to revert the formatting changes later."
     exit 1
 fi
 
-CLANGFORMAT="$(which clang-format-${CLANG_FORMAT_VER})"
-if [ "${CLANGFORMAT}" = "" ]; then
-    CLANGFORMAT="$(which clang-format)"
-    if ! "${CLANGFORMAT}" --version | grep -qEo "version ${CLANG_FORMAT_VER}\.[0-9]+"; then
-        echo "Could not find clang-format ${CLANG_FORMAT_VER}. ${CLANGFORMAT} is $(${CLANGFORMAT} --version | grep -Eo '[0-9\.]{5}' | head -n 1)."
-        exit 2
-    fi
-fi
-
-AUTOPEP8="$(which autopep8)"
-if ! "${AUTOPEP8}" --version 2>&1 | grep -qFo "autopep8 ${AUTOPEP8_VER} (pycodestyle: ${PYCODESTYLE_VER})"; then
-    echo "Could not find autopep8 ${AUTOPEP8_VER} with pycodestyle ${PYCODESTYLE_VER}"
-    echo "${AUTOPEP8} is $(${AUTOPEP8} --version 2>&1)"
+if ! hash pre-commit 2>/dev/null; then
+    echo "pre-commit command not found."
     exit 2
 fi
 
-find . \( -name '*.hpp' -o -name '*.cpp' -o -name '*.cu' -o -name '*.cuh' \) -not -path './libs/*' | xargs -r -n 5 -P 8 "${CLANGFORMAT}" -i -style=file || exit 3
-find . \( -name '*.py' -o -name '*.pyx' -o -name '*.pxd' \) -not -path './libs/*' | xargs -r -n 5 -P 8 "${AUTOPEP8}" --ignore=E266,W291,W293 --in-place --aggressive || exit 3
-find . -type f -executable ! -name '*.sh' ! -name '*.py' ! -name '*.sh.in' ! -name pypresso.cmakein -not -path './.git/*' | xargs -r -n 5 -P 8 chmod -x || exit 3
+pre-commit run --all-files
```

Comment:
I think dash and bash return the error code of the last command, so if this line fails due to a Python3.6-specific f-string (Ubuntu 16.04 uses Python3.5), the shell script will keep going to the next instructions and return 0, unless we add the following:
```suggestion
pre-commit run --all-files || exit 1
```

Reply:
Actually no, don't do that! `pre-commit` returns 1 when it touched a file.

Reply:
Is there an older version of `pre-commit` compatible with Python 3.5 that we can use?